src/aggregate.js

```diff
@@ -38,31 +42,29 @@ const config = require('./config');
  */
 module.exports = async function aggregate(collection, params) {
   params = _.defaults(await sanitizeParams(collection, params), { aggregation: [] });
-  const cursorQuery = generateCursorQuery(params);
-  const $sort = generateSort(params);
-
-  let index = _.findIndex(params.aggregation, (step) => !_.isEmpty(step.$match));
 
-  if (index < 0) {
-    params.aggregation.unshift({ $match: cursorQuery });
-    index = 0;
-  } else {
-    const matchStep = params.aggregation[index];
+  const $match = generateCursorQuery(params);
+  const $sort = generateSort(params);
+  const $limit = params.limit + 1;
 
-    params.aggregation[index] = {
-      $match: {
-        $and: [cursorQuery, matchStep.$match],
-      },
-    };
+  let addFields = [];
+  let cleanUp = [];
+  if (params.sortCaseInsensitive) {
+    addFields = [{ $addFields: { __lc: { $toLower: '$' + params.paginatedField } } }];
```

Comment:
So, we aren't using collation for sorting after all? Can you explain the rationale?

Reply:
There's not way to use collation without recreating the indices. Which would require duplicating all indices until we were sure that ALL queries used the collation. That would be an herculean task.

Using collations is still a possibility -- I've not removed those parameters.